packages/compass-schema/src/stores/store.js

```diff
@@ -217,6 +222,62 @@ const configureStore = (options = {}) => {
       }
     },
 
+    _calculateDepthByPath(input, paths) {
+      if (!input) {
+        return paths;
+      }
+      input.forEach(({types, fields, path, bsonType}) => {
+        const increment = bsonType === 'Array' ? 1 : 0;
+        const score = path.split('.').length + increment;
+        if (!paths[path] || paths[path] < score) {
+          paths[path] = score;
+        }
+        this._calculateDepthByPath(types ?? fields, paths);
+      });
+
+      return paths;
+    },
+
+    calculateSchemaDepth(schema) {
+      const response = this._calculateDepthByPath(schema.fields, {});
+      const values = Object.values(response);
+      return Math.max(...values, ...[0]);
+    },
+
+    _containsGeoData(input) {
+      let result = false;
+      if (!input) {
+        return result;
+      }
+      for (let i = 0; i < input.length; i++) {
+        const { path, values, types, fields } = input[i];
+        if (path.match(/type/) && intersection(MONGODB_GEO_TYPES, values).length > 0) {
```

Comment:
I’m a bit confused – when would `path` actually contain `type`? `path` is just the path to this sub-field in the document, right?

Reply:
from the schema-analysis response, the structure of geo document is as:
```
        {
                bsonType: 'Document',
                path: 'location',
                fields: [
                  {
                    name: 'coordinates',
                    path: 'location.coordinates',
                    types: [
                      {
                        bsonType: 'Array',
                        path: 'location.coordinates',
                        types: [
                          {
                            bsonType: 'Double',
                            path: 'location.coordinates',
                          },
                        ],
                      }
                    ],
                    type: 'Array',
                  },
                  {
                    name: 'type',
                    path: 'location.type',
                    types: [
                      {
                        bsonType: 'String',
                        path: 'location.type',
                        values: ['LineString', 'LineString', 'LineString', null, undefined, 123],
                      }
                    ],
                    type: 'String',
                  },
                ],
              },
```
we have to check for the values (which should be valid mdb geo options). values are contained in a child while its parent corresponds to `name: 'type'`. Either we have to pass in the type of parent or do this check on `path`. However, we can make this better by ensuring path ends with `type`.

Reply:
Right – I think you’d want `path.endsWith('.type')` here then :)